CI test failures after runner environment upgrades to Ubuntu 20.04

### Describe the bug

Since the upgrade of the Github Action runner environment to ubuntu 20.04 we're getting errors in the test suite 
```
E               paramiko.ssh_exception.SSHException: Error reading SSH protocol banner
```
https://github.com/aiidateam/aiida-core/pull/4683/checks?check_run_id=1774736504#step:9:2325

and
```
E           ValueError: ('Invalid private key', [_OpenSSLErrorWithText(code=67764350, lib=4, func=160, reason=126, reason_text=b'error:040A007E:rsa routines:RSA_check_key_ex:iqmp not inverse of q')])
```
https://github.com/aiidateam/aiida-core/pull/4683/checks?check_run_id=1774736504#step:9:453


It looks like paramiko is no longer able to parse the private SSH keys generated in the new OS

--- a/environment.yml
+++ b/environment.yml
@@ -24,7 +24,7 @@ dependencies:
 - kiwipy[rmq]~=0.7.1
 - numpy~=1.17
 - pamqp~=2.3
-- paramiko~=2.7
+- paramiko>=2.7.2,~=2.7
 - plumpy~=0.18.4
 - pgsu~=0.1.0
 - psutil~=5.6

--- a/requirements/requirements-py-3.6.txt
+++ b/requirements/requirements-py-3.6.txt
@@ -77,7 +77,7 @@ palettable==3.3.0
 pamqp==2.3
 pandas==0.25.3
 pandocfilters==1.4.2
-paramiko==2.7.1
+paramiko==2.7.2
 parso==0.6.2
 pathspec==0.8.0
 pexpect==4.8.0

--- a/requirements/requirements-py-3.7.txt
+++ b/requirements/requirements-py-3.7.txt
@@ -76,7 +76,7 @@ palettable==3.3.0
 pamqp==2.3
 pandas==0.25.3
 pandocfilters==1.4.2
-paramiko==2.7.1
+paramiko==2.7.2
 parso==0.6.2
 pathspec==0.8.0
 pexpect==4.8.0

--- a/requirements/requirements-py-3.8.txt
+++ b/requirements/requirements-py-3.8.txt
@@ -72,7 +72,7 @@ palettable==3.3.0
 pamqp==2.3
 pandas==0.25.3
 pandocfilters==1.4.2
-paramiko==2.7.1
+paramiko==2.7.2
 parso==0.6.2
 pexpect==4.8.0
 pg8000==1.13.2

--- a/setup.json
+++ b/setup.json
@@ -39,7 +39,7 @@
         "kiwipy[rmq]~=0.7.1",
         "numpy~=1.17",
         "pamqp~=2.3",
-        "paramiko~=2.7",
+        "paramiko~=2.7,>=2.7.2",
         "plumpy~=0.18.4",
         "pgsu~=0.1.0",
         "psutil~=5.6",

